Assets/Scripts/Game/PlayerCrush.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DaggerfallWorkshop.Game
+{
+    public class PlayerCrush : MonoBehaviour
+    {
+        private PlayerHeightChanger heightChanger;
+        private PlayerMotor playerMotor;
+        private CharacterController controller;
+        private float previousHeightHit;
+    
+	    void Start ()
+        {
+            heightChanger = GetComponent<PlayerHeightChanger>();
+            playerMotor = GetComponent<PlayerMotor>();
+            controller = GetComponent<CharacterController>();
+            previousHeightHit = 0f;
+	    }
+	
+	    void Update ()
+        {
+            if (GameManager.Instance.PlayerEntity.CurrentHealth < 1 
+                || GameManager.IsGamePaused)
+                return;
+
+            float distance;
+            if (!playerMotor.IsCrouching)
+                distance = (controller.height / 2f) - (controller.height * 0.1f);
+            else
+                distance = (controller.height / 2f);
+
+            Ray ray = new Ray(controller.transform.position, Vector3.up);
+            RaycastHit hit = new RaycastHit();
+            if (Physics.SphereCast(ray, controller.radius * 0.85f, out hit, distance))
+            {
+                if (hit.collider.GetComponent<MeshCollider>())
```

Comment:
Curious why grabbing MeshCollider<>() here. Is this to test if player is being crushed under actual scene geometry compared to say an enemy walking off a ledge above and landing on player's head?

Reply:
LOL. 🤣 earlier when I was using the CapsuleColider as a form of detection, I was getting forced to crouch when I was walking over dead enemy corpses and loot piles.  And I kept it in the code when i transferred and modified it for the update method. I suppose it's still possible when an enemy dies on some ledge, if you walk up to the ledge you might trigger a collision.

Reply:
Yep, this is a reasonable solution to the problem of dealing with non-level geometry like enemies and loot piles as they don't carry mesh colliders. Well done.

My other concern is what happens if player manages to crouch and crawl in somewhere very tight (e.g. on top of a shelf very close to the ceiling). They would potentially be both grounded and within range to trigger a crush. That's something we'd have to test. I'll stick it to a branch and we'll go from there.

Reply:
This works very well so far! There's a spot in Mantellan Crux where player can crawl under a low shelf slightly lower than crouching head height., But with the tolerances you've put in place it all just works as expected. The player literally has to stand under a lift and let it lower all the way down onto them, which is perfect.

Also did some profiling, no real impact to physics system. I think Unity perform these kinds of tests very efficiently and continuously under the hood. So no problems with that either.

I'll play with it a bit more and try to break it. Will merge to master soon. Good work! 👍